flink-core/src/main/java/org/apache/flink/metrics/groups/ProxyMetricGroup.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.metrics.groups;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.metrics.Counter;
+import org.apache.flink.metrics.Gauge;
+import org.apache.flink.metrics.Histogram;
+import org.apache.flink.metrics.MetricGroup;
+import org.apache.flink.util.Preconditions;
+
+/**
+ * Metric group which forwards all registration calls to its parent metric group.
+ *
+ * @param <P> Type of the parent metric group
+ */
+@Internal
+public class ProxyMetricGroup<P extends MetricGroup> implements MetricGroup {
+	private final P parentMetricGroup;
+
+	public ProxyMetricGroup(P parentMetricGroup) {
+		this.parentMetricGroup = Preconditions.checkNotNull(parentMetricGroup);
+	}
+
+	@Override
+	public final void close() {
+		parentMetricGroup.close();
```

Comment:
Hmm you're right, it is a bit tricky. You don't want to close the complete parent metric group. But I guess you would want to unregister the metrics registered by the `IOMetricGroup` if you call `IOMetricGroup#close`. However, this is not really possible at the moment. I will remove the `parent.close` call then.

I agree that `close` and `isClosed` should only be used internally and not being exposed to the user. This could be a follow-up task.